test/common/upstream/upstream_impl_test.cc

```diff
@@ -2813,6 +2813,49 @@ TEST_F(ClusterInfoImplTest, DefaultConnectTimeout) {
   EXPECT_EQ(std::chrono::seconds(5), cluster->info()->connectTimeout());
 }
 
+TEST_F(ClusterInfoImplTest, MaxConnectionDurationTest) {
+  const std::string yaml_default_max_connection_duration = R"EOF(
```

Comment:
I was expecting to see a test using simulated time, where we advance time forward past the timeout and then verify that the connections get properly closed.

Reply:
I updated my new tests to start using a derived ConnPoolImplDispatcherBaseTest that uses a real (non-mock) dispatcher and simulated time. I left the existing mock dispatcher to avoid modifying existing tests, especially the ones that derive from this base test.

Reply:
thanks!